network/p2p/throttled_handler.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+var _ Handler = (*ThrottledHandler)(nil)
+
+type ThrottledHandler struct {
+	Handler
```

Comment:
We're only throttling `AppRequest` because it requires a response. I thought it made sense for us to just allow any `AppGossip`/local requests on `CrossChainAppRequest`.